lib/features/mailbox/data/network/mailbox_api.dart

```diff
@@ -193,11 +197,21 @@ class MailboxAPI {
 
       finalResult = currentSetMailboxInvocations
         .map((currentInvocation) => response.parse(currentInvocation.methodCallId, SetMailboxResponse.deserialize))
-        .map((response) => response?.destroyed?.isNotEmpty ?? false)
-        .every((element) => element == true);
+        .map((response) {
+          if(response?.notDestroyed != null) {
+            deleteMailboxErrors.addAll(response!.notDestroyed!);
```

Comment:
`response` here is non-null. please carefully

Reply:
check at line 201

Reply:
it is not a check for `response`. Still risk when using `response!`